### PR TITLE
Feature/open existing contact

### PR DESCRIPTION
**Pull request by cmasus**
Return contact photo when opening or editing a contact in native phonebook.
Adding openExistingContact method.
Adding writePhotoToPath method in android.

### CODE REVIEW

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java [hunk 1]
```diff
@@ -840,4 +907,58 @@ private int mapStringToPostalAddressType(String label) {
...
+            updateContactCallback.invoke("No data", null);
+            updateContactCallback = null;
+            return;
+        }
+
+        try {
+            Uri contactUri = data.getData();
+
+            if (contactUri == null) {
+                updateContactCallback.invoke(null, null); // something was wrong
```
Comment (rhoglund) [inline android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]:
```suggestion
                updateContactCallback.invoke("Error wrong data. No content uri found!", null); // something was wrong
```

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java [hunk 2]
```diff
@@ -840,4 +907,58 @@ private int mapStringToPostalAddressType(String label) {
...
+        }
+
+        if (resultCode != Activity.RESULT_OK) {
+            updateContactCallback.invoke(null, null); // user probably pressed cancel
+            updateContactCallback = null;
+            return;
+        }
+
+        if (data == null) {
+            updateContactCallback.invoke("No data", null);
```
Comment (rhoglund) [inline android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]:
```suggestion
            updateContactCallback.invoke("Error received activity result with no data!", null);
```

File: index.d.ts [hunk 3]
```diff
@@ -3,12 +3,14 @@ declare module 'react-native-contacts' {
...
   export function addContact(contact: Contact, callback: (error?: any) => void): void;
-  export function openContactForm(contact: Contact, callback: (error?: any) => void): void;
+  export function openContactForm(contact: Contact, callback: (error: any, contact: Contact) => void): void;
+  export function openExistingContact(contact: Contact, callback: (error: any, contact: Contact) => void): void;
   export function updateContact(contact: Contact, callback: (error?: any) => void): void;
   export function deleteContact(contact: Contact, callback: (error?: any) => void): void;
   export function getContactsMatchingString(str: string, callback: (error: any, contacts: Contact[]) => void): void;
   export function checkPermission(callback: (error: any, result: 'authorized' | 'denied' | 'undefined') => void): void;
   export function requestPermission(callback: (error: any, result: 'authorized' | 'denied' | 'undefined') => void): void;
+  export function writePhotoToPath(contactId: string, file: string, callback: (error: any, result: boolean) => void): void;
```
Comment (rhoglund) [inline index.d.ts:13]: This method has no implementation on iOS. Write an implementation that will just return an error ('Error method not implemented on iOS')

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java (hunk 1 above)
Comment (cmasus) [reply android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]: Changed to suggestion.

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java (hunk 2 above)
Comment (cmasus) [reply android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]: Changed to suggestion.

**Pull request merged**